sycl/doc/extensions/proposed/sycl_ext_oneapi_group_load_store_extended.asciidoc

```diff
@@ -0,0 +1,506 @@
+* `InputT` must be convertible to value type of `OutputIteratorT`.
+
+_Effects_: Stores `ElementsPerWorkItem` elements from `in` span to `out_iter`
+using the `g` group object.
+
+Properties may specify xref:data_placement[data placement].
+Default placement is a blocked one:
+`out_iter[g.get_local_linear_id() * ItemsPerWorkItem + i]` = `in[i];`
+in striped case:
+`out_iter[g.get_local_linear_id() + g.get_local_linear_range() * i]` = `in[i];`
+for `i` between `0` and `ItemsPerWorkItem`.
+
+
+=== Data Placement
+
+anchor:data_placement[]
+
+To specify a correct data placement for placing of resulting data
+there is a enum
+(proposed in link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]):
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+// Properties:
+enum class group_algorithm_data_placement {
+  blocked,
+  striped
+};
+
+}
+----
+
+Data placement layout example on group_load:
+
+* ElementsPerWorkItem = 4
+* 3 work-items in the group
+* input is: in_iter[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11}
+
+Consider 2 layouts:
+
+1.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::blocked`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 1, 2, 3}
+|1
+|{4, 5, 6, 7}
+|2
+|{8, 9, 10, 11}
+|===
+
+2.`sycl::ext::oneapi::experimental::group_algorithm_data_placement::striped`.
+
+|===
+|Work-item id|Output stored in private fixed-size array
+
+|0
+|{0, 3, 6, 9}
+|1
+|{1, 4, 7, 10}
+|2
+|{2, 5, 8, 11}
+|===
+
+This extention adds a property that satisfies
+link:../experimental/sycl_ext_oneapi_properties.asciidoc[SYCL Properties Extension]
+requirements to identify data_placement similar to
+link:https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_oneapi_group_sort.asciidoc[`group_sort extention`]
+`input_data_placement` and `output_data_placement`:
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental::property
+{
+    template<group_algorithm_data_placement type>
+    struct data_placement;
+}
+----
+
+Specifies data layout used in group_load/store for `sycl::vec` or fixed-size
+arrays functions.
+
+Example:
+`group_load(g, input, output_span, properties<data_placement<blocked>>{});`
+
+=== Usage Example
+
+Example shows the simplest case without local memory usage of blocked load
+of global memory from `input` to the private array `data` and store it back to
+`output`
+
+[source,c++]
+----
+namespace sycl_exp = sycl::ext::oneapi::experimental;
+
+constexpr std::size_t block_size = 32;
+constexpr std::size_t items_per_thread = 4;
+constexpr std::size_t block_count = 2;
+constexpr std::size_t size = block_count * block_size * items_per_thread;
+
+sycl::queue q;
+sycl::usm_allocator<T, sycl::usm::alloc::device, 64> allocator(q);
+std::vector<T, decltype(allocator)> input(size, allocator);
+std::vector<T, decltype(allocator)> output(size, allocator);
```

Comment:
We don't support `usm_allocator` with `usm::alloc::device` memory.  This example also captures `std::vector` as a kernel argument which is not officially supported as an argument type.  I think it would be simpler to just use `malloc_device` here.

Reply:
Fixed, thank you